.github/workflows/master.yml

```
@@ -35,6 +38,28 @@ jobs:
           token: ${{ secrets.GITHUB_TOKEN }}
           args: --all-features -- -D warnings
 
+  solana-build: 
+    name: Cargo Build using bpf (solana env)
+    runs-on: ubuntu-latest
+    steps: 
+      - uses: actions/checkout@v3
+      - name: Install Rust
+        uses: dtolnay/rust-toolchain@master
+        with:
+          toolchain: stable
+          components: rustfmt
+
+      - name: Install Solana CLI
+        run: curl -sSfL https://release.solana.com/$SOLANA_VERSION/install | sh
+        shell: bash
+
+      - name: Adding Solana CLI to the path
+        run: echo "/home/runner/.local/share/solana/install/active_release/bin" >> $GITHUB_PATH
+        shell: bash
+      
+      - name: Run build-bpf
+        run: cargo build-bpf
```

Do you need to specify project here?

```suggestion
        run: cargo build-bpf -p solana-ibc
```

---

`cargo build-bpf` doesnt take `-p` as argument. But running `cargo build-bpf` fails on local so it should fail here too.

---

How does it know which package to build? Do we need to `cd solana/solana-ibc` or something?

---

its like `cargo build`, it picks it up from the cargo workspace.

---

> But running cargo build-bpf fails on local so it should fail here too.

For me `cargo build-bpf` and `cargo build-bpf -- -p solana-ibc` work fine.

For specifying project, it’s actually `cargo build-bpf -- -p solana-ibc`. We care only about that project so no point trying to build others using bpf.

Lastly, should we use `build-sbf` instead?

---

Did the build fail for you in local?

> Lastly, should we use build-sbf instead?

Yes, we should since the bpf is deprecated.

---

> Did the build fail for you in local?

No, it works just fine:

```
$ cargo build-sbf -- -p solana-ibc
Compiling cfg-if v1.0.0
...
    Finished release [optimized] target(s) in 49.56s
```

---

Thats strange, because it fails for me. So i had fixed it in another PR which is merged with the master. But dont know why its not failing here or locally for u? Is it smtg wrong on my local?

---

Oh, OK, I haven’t tested this PR and yes it does fail:

```
$ cargo build-sbf -- -p solana-ibc
...
error[E0412]: cannot find type `Vec` in this scope
   --> common/lib/src/hash.rs:186:29
    |
186 |     pub(super) struct Inner(Vec<u8>);
    |                             ^^^ not found in this scope
    |
help: consider importing one of these items
    |
186 |     use alloc::vec::Vec;
    |
186 |     use borsh::maybestd::vec::Vec;
    |

error[E0425]: cannot find value `bytes` in this scope
   --> common/lib/src/hash.rs:196:43
    |
196 |             solana_program::hash::hashv(&[bytes]).to_bytes()
    |                                           ^^^^^ not found in this scope

error[E0277]: the trait bound `Inner: Default` is not satisfied
   --> common/lib/src/hash.rs:211:20
    |
210 | #[derive(Default)]
    |          ------- in this derive macro expansion
211 | pub struct Builder(builder::Inner);
    |                    ^^^^^^^^^^^^^^ the trait `Default` is not implemented for `Inner`
    |
    = note: this error originates in the derive macro `Default` (in Nightly builds, run with -Z macro-backtrace for more info)
...
```

---

right right, then it means there is smtg wrong with the actions.

---

Nothing is wrong with actions. Since the issue is resolved in master, the CI was passing.